pyccel/parser/semantic.py

```diff
@@ -3859,6 +3881,10 @@ def _visit_ClassDef(self, expr):
             m_name = method.name
             if m_name == '__init__':
                 self._visit(method)
+                if cls.interfaces:
+                    errors.report("Pyccel does not support interface constructor", symbol=m_name,
+                        bounding_box=(self._current_fst_node.lineno, self._current_fst_node.col_offset),
+                        severity='fatal')
```

Comment:
`_current_fst_node` was renamed in a recent PR. But why not avoid this problem by providing the method directly:

```suggestion
                    errors.report("Pyccel does not support interface constructor",
                        symbol=method, severity='fatal')
```